src/libraries/System.Memory/src/System/Buffers/SequenceReader.cs

```diff
@@ -141,6 +141,11 @@ public SequenceReader(ReadOnlySequence<T> sequence)
             // If offset doesn't fall inside current segment move to next until we find correct one
             if ((CurrentSpanIndex + offset) <= CurrentSpan.Length - 1)
             {
+                // Sum CurrentSpanIndex + offset could overflow as is but the value of offset should be very large
+                // because we check Remaining <= offset above so to overflow we should have a ReadOnlySequence close to 8 exabytes
+                Debug.Assert(CurrentSpanIndex + offset >= 0);
```

Comment:
nit: move this before the check (i.e. do this, with the comment. before line 141). Adding the check within the if-check is less helpful.

Reply:
ok my idea was check only in case of overflow(negative) and satisfied condition above, btw I'll do it, thank's